src/main/java/duke/views/cli/strategies/MultiType.java

```diff
@@ -0,0 +1,238 @@
+package duke.views.cli.strategies;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import duke.constants.Constants;
+import duke.domain.Deadline;
+import duke.domain.Event;
+import duke.domain.Task;
+import duke.domain.Todo;
+import duke.shared.DukeException;
+import duke.shared.DukeException.ExceptionCode;
+
+/**
+ * A responder that is able to handle tasks with CRUD functionality.
+ */
+public class MultiType extends RespondWith {
+
+    private final String list = "list";
+    private final String done = "done";
+    private final String todo = "todo";
+    private final String deadline = "deadline";
+    private final String event = "event";
+    private final String delete = "delete";
+    private final String on = "on";
+    private final String find = "find";
+
+    private final List<Task> userTasks;
+
+    /**
+     * Creates a responder that handles multiple types of tasks.
+     */
+    public MultiType() {
+        userTasks = new ArrayList<>();
+        commands.put(list, this::listString);
+        commands.put(done, this::markDone);
+        commands.put(todo, this::addTodo);
+        commands.put(deadline, this::addDeadline);
+        commands.put(event, this::addEvent);
+        commands.put(delete, this::deleteTask);
+        commands.put(on, this::occurringOn);
+        commands.put(find, this::findTask);
+    }
+
+    @Override
+    public Task rehydrateFromString(String s) {
+        String[] fields = s.split(Constants.Storage.PERSISTENCE_SEPARATOR_REGEX, -1);
+        for (int i = 0; i < fields.length; i++) {
+            fields[i] = fields[i].strip();
+        }
+        String typeString = fields[0];
+        Task task = null;
+        switch (typeString) {
+        case Todo.TYPE_STRING: {
```

Comment:
I think the switch statement will still work fine without the brackets {} around each case 😄.

Reply:
Yep, they can be removed.